Modules/FindNetCDF.cmake

```diff
@@ -183,7 +183,7 @@ else()
 endif()
 
 if(NetCDF_PARALLEL)
-  find_package(MPI REQUIRED)
+  find_package(MPI)
```

Comment:
If `NetCDF_PARALLEL` is `True` and MPI is not found, wouldn't the linkages of the netcdf library be incomplete?

Reply:
In the case described in #50, this works.
On the other hand, what is the reasoning for the second check for the MPI components then?
https://github.com/NOAA-EMC/CMakeModules/blob/12f5cc2cc3cd16cb8af1ecd4863cf55025691717/Modules/FindNetCDF.cmake#L249

Reply:
@kgerheiser also said, this would be fine: https://github.com/NOAA-EMC/CMakeModules/issues/50#issuecomment-862821740